### PR TITLE
Adds Token Store

### LINKED ISSUE #4049: Implement `read-only` `TokenStore` implementation supporting key access
Implement a read-only TokenStore that supports key access

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #4049

- Adds `TokenStore` that returns `TokenMetadata` including all keys on the token

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+								Optional<JKey> kycKey,
+								Optional<JKey> wipeKey,
+								Optional<JKey> freezeKey,
+								Optional<JKey> supplyKey,
+								Optional<JKey> feeScheduleKey,
+								Optional<JKey> pauseKey,
+								boolean hasRoyaltyWithFallback,
+								EntityId treasury) {}
+
+	public record TokenMetaOrReason(TokenMetadata metadata, ResponseCodeEnum failureReason) {
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java]: Maybe `TokenResult`, `TokenOperationResult`?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java]: Used `OrReason` following the same naming as `KeyorLookupFailureReason` in `AccountStore`

**Review by mhess-swl — APPROVED** [review]

**Comment by kimbor** [comment]
It's a bit of a nit, but if the name of the `TokenMetaOrLookUpFailureReason` record is modeled after `KeyOrLookupFailureReason` then the case of Lookup should match, i.e. lowercase the 'U'.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            Optional<JKey> kycKey,
+            Optional<JKey> wipeKey,
+            Optional<JKey> freezeKey,
+            Optional<JKey> supplyKey,
+            Optional<JKey> feeScheduleKey,
+            Optional<JKey> pauseKey,
+            boolean hasRoyaltyWithFallback,
+            EntityId treasury) {}
+
+    public record TokenMetaOrLookUpFailureReason(TokenMetadata metadata, ResponseCodeEnum failureReason) {
```
Comment (kimbor) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java:59]: It's a bit of a nit, but if the name of the TokenMetaOrLookUpFailureReason record is modeled after KeyOrLookupFailureReason then the case of Lookup should match, i.e. lowercase the 'U'.

**Review by kimbor — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/TokenStore.java]: Thanks Kim .. fixed it !

**Pull request merged**